src/javascript/app/ng-wig/ng-wig.component.spec.js

```diff
@@ -0,0 +1,263 @@
+describe('component: ngWig', () => {
+    let component;
+    let controller;
+    let scope;
+    let content;
+    let options;
+    let buttons;
+    let beforeExecCommand;
+    let afterExecCommand;
+    let element;
+    let attrs;
+    let pasteSpy;
+    let beforeExecSpy;
+    let afterExecSpy;
+    let mocks;
+    let window;
+    let compile;
+
+    beforeEach(module('ngWig'));
+
+    beforeEach(inject((_$componentController_, _$rootScope_, _$window_, _$compile_, _ngWigToolbar_) => {
+        window = _$window_;
+        compile = _$compile_;
+
+        scope = _$rootScope_.$new();
+
+        content = 'Fake text';
+        element = angular.element('<div></div>');
+        attrs = { $observe: () => {} };
+        options = ['Option 1', 'Option 2'];
+        buttons = 'button1,button2';
+        pasteSpy = jasmine.createSpy('pasteSpy');
+        beforeExecSpy = jasmine.createSpy('beforeExecSpy');
+        afterExecSpy = jasmine.createSpy('afterExecSpy');
+
+        mocks = {
+            list1: {title: 'Unordered List', command: 'insertunorderedlist', styleClass: 'list-ul'},
+            list2: {title: 'Ordered List', command: 'insertorderedlist', styleClass: 'list-ol'},
+            bold: {title: 'Bold', command: 'bold', styleClass: 'bold'},
+            italic: {title: 'Italic', command: 'italic', styleClass: 'italic'},
+            link: {title: 'Link', command: 'createlink', styleClass: 'link'}
+        };
+
+        spyOn(_ngWigToolbar_, 'getToolbarButtons').and.returnValue(mocks);
+        spyOn(window.getSelection(), 'removeAllRanges');
+
+        component = _$componentController_('ngWig', 
+                                            { $scope: scope, $element: element, $attrs: attrs }, 
+                                            { content: content, options: options, buttons: buttons, onPaste: pasteSpy, beforeExecCommand: beforeExecSpy, afterExecCommand: afterExecSpy }
+                                            );
+    }));
+
+    function getCompiledElement(template) {
+        let element = angular.element(template || '<ng-wig ng-model="text1"><ng-wig>');
+        let compiledElement = compile(element)(scope);
+        
+        scope.$digest();
+
+        controller = element.controller('ngWig');
+        
+        return compiledElement;
+    }
+
+    it('should expose content', () => {
```

Comment:
I like that but I have made second thoughts. I have read that good unit tests provide good documentation and since there is no yet an official one, we could keep tests separate.